modules/loggers/mod_log_config.c

```diff
@@ -1666,13 +1781,279 @@ static ap_log_writer *ap_log_set_writer(ap_log_writer *handle)
+        }
+
+        lfdj->total_len += add_str(strs, strl, "\"");
+        lfdj->total_len += add_str(strs, strl, attribute_name);
+        lfdj->total_len += add_str(strs, strl, "\":");
+
+        /* if the current tag has arguments, i.e. "i" (requestHeaders)
+         * create a sub object with items[i].arg as key and lfd->portions[i] values.
+         *
+         * as multiple same tags with different arguments can exist,
+         * the log_format_item is sorted before, so we can easily check for
+         * gruppenwechsel
+         */
+        if(items[i].arg != NULL && strlen(items[i].arg) > 0) {
+            /* start sub object */
+            lfdj->total_len += add_str(strs, strl, "{");
+            for (j = i; j < lfd->nelts; ++j) {
+                if(items[j].tag == NULL) {
+                    continue;
+                }
+                if(strcmp(items[j].tag, items[i].tag) != 0) {
+                    i = j; /* -1 ?*/
+                    break;
+                }
+                attribute_name = ap_escape_logjson(r->pool, items[i].arg, NULL, 0);
+
+                lfdj->total_len += add_str(strs, strl, "\"");
+                lfdj->total_len += add_str(strs, strl, attribute_name);
+                lfdj->total_len += add_str(strs, strl, "\":");
+
+                if (!lfd->portions[i]) {
+                    lfdj->total_len += add_str(strs, strl, "null");
+                }
+                else if (!*lfd->portions[i]) {
+                    lfdj->total_len += add_str(strs, strl, "\"\"");
+                }
+                else {
+                    lfdj->total_len += add_str(strs, strl, "\"");
+                    attribute_value = ap_escape_logjson(r->pool, lfd->portions[i], NULL, 0);
+                    lfdj->total_len += add_str(strs, strl, attribute_value);
+                    lfdj->total_len += add_str(strs, strl, "\"");
+                }
+                lfdj->total_len += add_str(strs, strl, ",");
+            }
+            /* remove last "," */
+            apr_array_pop(strs);
+            lfdj->total_len -= *(int *)apr_array_pop(strl);
+
+            /* end sub object */
+            lfdj->total_len += add_str(strs, strl, "},");
+            continue;
+        }
+
+        if (!lfd->portions[i]) {
+            lfdj->total_len += add_str(strs, strl, "null");
+        }
+        else if (!*lfd->portions[i]) {
+            lfdj->total_len += add_str(strs, strl, "\"\"");
+        }
+        else {
+            lfdj->total_len += add_str(strs, strl, "\"");
+            attribute_value = ap_escape_logjson(r->pool, lfd->portions[i], NULL, 0);
+            lfdj->total_len += add_str(strs, strl, attribute_value);
+            lfdj->total_len += add_str(strs, strl, "\"");
+        }
+        lfdj->total_len += add_str(strs, strl, ",");
+    }
+    /* replace last ',' with '}' */
+    apr_array_pop(strs);
+    lfdj->total_len -= *(int *)apr_array_pop(strl);
```

Comment:
Likewise here with some `found_i`.

Reply:
should be fixed with latest commit which implements omit null values